### ISSUE TITLE
can't remap MENU-KEY_XXX to anther key

### ISSUE THREAD

**Issue by pzzxqj** [k0kubun/xremap#349]
```yml
keymap:
  - name: dirction
    remap:
      # COMPOSE: reserved

      COMPOSE-H: LEFT
      COMPOSE-J: DOWN
      COMPOSE-K: UP
      COMPOSE-L: RIGHT

      C-M-H: LEFT
      C-M-J: DOWN
      C-M-K: UP
      C-M-L: RIGHT

      COMPOSE-S: HOME
      COMPOSE-D: PAGEDOWN
      COMPOSE-F: PAGEUP
      COMPOSE-G: END
```

log:
```
Selecting devices from the following list:
------------------------------------------------------------------------------
/dev/input/event0 : Power Button
/dev/input/event1 : Power Button
/dev/input/event10: HDA NVidia HDMI/DP,pcm=8
/dev/input/event11: HDA NVidia HDMI/DP,pcm=9
/dev/input/event12: HD-Audio Generic Front Mic
/dev/input/event13: HD-Audio Generic Rear Mic
/dev/input/event14: HD-Audio Generic Line
/dev/input/event15: HD-Audio Generic Line Out Front
/dev/input/event16: HD-Audio Generic Line Out Surround
/dev/input/event17: HD-Audio Generic Line Out CLFE
/dev/input/event18: HD-Audio Generic Front Headphone
/dev/input/event2 : USB Optical Mouse
/dev/input/event3 : HID 026d:0002
/dev/input/event4 : HID 026d:0002 Consumer Control
/dev/input/event5 : HID 026d:0002 Keyboard
/dev/input/event6 : MSI PRO CARBON   
/dev/input/event7 : PC Speaker
/dev/input/event8 : HDA NVidia HDMI/DP,pcm=3
/dev/input/event9 : HDA NVidia HDMI/DP,pcm=7
------------------------------------------------------------------------------
Selected keyboards automatically since --device options weren't specified:
------------------------------------------------------------------------------
/dev/input/event3 : HID 026d:0002
/dev/input/event5 : HID 026d:0002 Keyboard
------------------------------------------------------------------------------
[2023-10-01T12:22:11Z DEBUG xremap::event_handler] => 0: KEY_ENTER
[2023-10-01T12:22:11Z DEBUG xremap::action_dispatcher] 0: KEY_ENTER
[2023-10-01T12:22:13Z DEBUG xremap::event_handler] => 1: KEY_COMPOSE
[2023-10-01T12:22:13Z DEBUG xremap::action_dispatcher] 1: KEY_COMPOSE
[2023-10-01T12:22:13Z DEBUG xremap::event_handler] => 2: KEY_COMPOSE
[2023-10-01T12:22:13Z DEBUG xremap::action_dispatcher] 2: KEY_COMPOSE
[2023-10-01T12:22:13Z DEBUG xremap::event_handler] => 2: KEY_COMPOSE
[2023-10-01T12:22:13Z DEBUG xremap::action_dispatcher] 2: KEY_COMPOSE
[2023-10-01T12:22:14Z DEBUG xremap::event_handler] => 2: KEY_COMPOSE
[2023-10-01T12:22:14Z DEBUG xremap::action_dispatcher] 2: KEY_COMPOSE
[2023-10-01T12:22:14Z DEBUG xremap::event_handler] => 2: KEY_COMPOSE
[2023-10-01T12:22:14Z DEBUG xremap::action_dispatcher] 2: KEY_COMPOSE
[2023-10-01T12:22:14Z DEBUG xremap::event_handler] => 2: KEY_COMPOSE
[2023-10-01T12:22:14Z DEBUG xremap::action_dispatcher] 2: KEY_COMPOSE
[2023-10-01T12:22:14Z DEBUG xremap::event_handler] => 2: KEY_COMPOSE
[2023-10-01T12:22:14Z DEBUG xremap::action_dispatcher] 2: KEY_COMPOSE
[2023-10-01T12:22:14Z DEBUG xremap::event_handler] => 2: KEY_COMPOSE
[2023-10-01T12:22:14Z DEBUG xremap::action_dispatcher] 2: KEY_COMPOSE
[2023-10-01T12:22:14Z DEBUG xremap::event_handler] => 1: KEY_J
[2023-10-01T12:22:14Z DEBUG xremap::action_dispatcher] 1: KEY_J
[2023-10-01T12:22:14Z DEBUG xremap::event_handler] => 0: KEY_J
[2023-10-01T12:22:14Z DEBUG xremap::action_dispatcher] 0: KEY_J
[2023-10-01T12:22:14Z DEBUG xremap::event_handler] => 1: KEY_K
[2023-10-01T12:22:14Z DEBUG xremap::action_dispatcher] 1: KEY_K
[2023-10-01T12:22:14Z DEBUG xremap::event_handler] => 0: KEY_K
[2023-10-01T12:22:14Z DEBUG xremap::action_dispatcher] 0: KEY_K
[2023-10-01T12:22:14Z DEBUG xremap::event_handler] => 1: KEY_L
[2023-10-01T12:22:14Z DEBUG xremap::action_dispatcher] 1: KEY_L
[2023-10-01T12:22:14Z DEBUG xremap::event_handler] => 0: KEY_L
[2023-10-01T12:22:14Z DEBUG xremap::action_dispatcher] 0: KEY_L
[2023-10-01T12:22:15Z DEBUG xremap::event_handler] => 1: KEY_H
[2023-10-01T12:22:15Z DEBUG xremap::action_dispatcher] 1: KEY_H
[2023-10-01T12:22:15Z DEBUG xremap::event_handler] => 0: KEY_H
[2023-10-01T12:22:15Z DEBUG xremap::action_dispatcher] 0: KEY_H
[2023-10-01T12:22:15Z DEBUG xremap::event_handler] => 0: KEY_COMPOSE
[2023-10-01T12:22:15Z DEBUG xremap::action_dispatcher] 0: KEY_COMPOSE
[2023-10-01T12:22:16Z DEBUG xremap::event_handler] => 1: KEY_COMPOSE
[2023-10-01T12:22:16Z DEBUG xremap::action_dispatcher] 1: KEY_COMPOSE
[2023-10-01T12:22:16Z DEBUG xremap::event_handler] => 0: KEY_COMPOSE
[2023-10-01T12:22:16Z DEBUG xremap::action_dispatcher] 0: KEY_COMPOSE
[2023-10-01T12:22:19Z DEBUG xremap::event_handler] => 1: KEY_COMPOSE
[2023-10-01T12:22:19Z DEBUG xremap::action_dispatcher] 1: KEY_COMPOSE
[2023-10-01T12:22:20Z DEBUG xremap::event_handler] => 1: KEY_K
[2023-10-01T12:22:20Z DEBUG xremap::action_dispatcher] 1: KEY_K
[2023-10-01T12:22:20Z DEBUG xremap::event_handler] => 0: KEY_K
[2023-10-01T12:22:20Z DEBUG xremap::action_dispatcher] 0: KEY_K
[2023-10-01T12:22:20Z DEBUG xremap::event_handler] => 0: KEY_COMPOSE
[2023-10-01T12:22:20Z DEBUG xremap::action_dispatcher] 0: KEY_COMPOSE
```

**Comment by k0kubun** [comment]
What if you add the following lines to your config?

```yml
virtual_modifiers:
  - COMPOSE
```

**Comment by pzzxqj** [comment]
The log shows compose when I press menu key. By the way, I can disable the menu key by remapping `compose` to `reserved`

**Comment by k0kubun** [comment]
I have no idea what you're trying to do. What do you want to fix?

**Comment by k0kubun** [comment]
Let's say "can't remap MENU-KEY_XXX to another key" is what you want. Then,

> The log shows compose when I press menu key.

This problem is not related to xremap. It's just what the kernel (or its evdev) does. If you think that is a problem, consider reporting it to the kernel or the keyboard manufacturer. xremap is only remapping your input, which was actually "compose".

So the problem (to xremap) is actually "can't remap COMPOSE-KEY_XXX to another key". Neither COMPOSE nor MENU is a modifier, so it's currently intentional that it's not remapped unless you add the key to `virtual_modifiers`. You somehow ignored my comment https://github.com/k0kubun/xremap/issues/349#issuecomment-1742192406, but that's what your config would need to fix that issue.

**Comment by pzzxqj** [comment]
Yeah, I want to use `menu` or `compose` as modifier.

Thanks for the suggestion, I used the 'virtual_modifiers` method and it now works.

**Comment by Vaisakhkm2625** [comment]
is there any mistake with what i have done?
```
virtual_modifiers:
  - KEY_MENU

modmap:
  - name: Global
    remap:
      capsLock:
        held: leftctrl
        alone: esc
        alone_timeout_millis: 150 

      KEY_MENU: KEY_LEFTMETA

```

**Comment by k0kubun** [comment]
try ctrl_l instead of leftctrl?

**Comment by k0kubun** [comment]
also meta_l

**Comment by k0kubun** [comment]
well, that might be fine. what was the error you got? it's hard to debug anything without reading error messages.

**Comment by Vaisakhkm2625** [comment]
sorry i should have been more specific

```
virtual_modifiers:
  - KEY_MENU

modmap:
  - name: Global
    remap:
      KEY_MENU: KEY_LEFTMETA
      #KEY_MENU: KEY_SUPER_L # this errors (wev showed Super_L)

```

what i wanted was to remap menu key to super key (which i thought might be leftmeta [from this names])(https://github.com/emberian/evdev/blob/1d020f11b283b0648427a2844b6b980f1a268221/src/scancodes.rs#L26-L572) above runs without errors, but Menu key is not acting as meta

**Comment by k0kubun** [comment]
That's not how `virtual_modifiers` is supposed to be used. You never need to remap a virtual modifier like `KEY_MENU: KEY_LEFTMETA` because it's already a modifier without remapping to an actual modifier, `LEFTMETA`. You can use `KEY_MENU-something: xxx` without remapping `KEY_MENU` to anything else once you declare it in `virtual_modifiers`. If you actually want to use `KEY_MENU` as `KEY_LEFTMETA`, simply remove it from `virtual_modifiers`.

**Comment by Vaisakhkm2625** [comment]
TL;DR: _should have used evtest instead of wev/xev or ran xremap in debug mode.. acutal name of key was compose, not menu._


Extremely sorry for wasting your time, actually i was trying without virtual_modifiers and then only i found this issue and decided to try virtual modifiers

Actual issue was key i am trying to remap is not called Menu, But Compose. 

evtest report
```
^[[29~Event: time 1707155410.691285, type 4 (EV_MSC), code 4 (MSC_SCAN), value 70065
Event: time 1707155410.691285, type 1 (EV_KEY), code 127 (KEY_COMPOSE), value 0
Event: time 1707155410.691285, -------------- SYN_REPORT ------------
```

all the issue was caused by `wev`(xorg.xev too) reporting wrong keycodes and as `scancodes` shown by `evdev` is different from `keycodes`, which is shown by wev/xev. [more about this](https://github.com/bevyengine/bevy/discussions/2386) 

i got the idea to use wev from this discussion - https://github.com/k0kubun/xremap/discussions/356 
now only i noticed your answer of running xremap in debug mode.
Thanks for helping me out...
